src/app/Http/Controllers/Operations/FetchAjaxOperation.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace Backpack\CRUD\app\Http\Controllers\Operations;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Config;
+use Illuminate\Support\Facades\Route;
+
+trait FetchAjaxOperation
+{
+    /**
+     * Define which routes are needed for this operation.
+     *
+     * @param string $segment    Name of the current entity (singular). Used as first URL segment.
+     * @param string $routeName  Prefix of the route name.
+     * @param string $controller Name of the current CrudController.
+     */
+    protected function setupFetchAjaxOperationRoutes($segment, $routeName, $controller)
+    {
+        if ($this->crud->hasOperationSetting('ajaxEntities')) {
+            $ajaxEntities = $this->crud->getOperationSetting('ajaxEntities');
+            foreach ($ajaxEntities as $relatedSegment => $entiyToFetch) {
+                $routeSegment = isset($entiyToFetch['routeSegment']) ? $entiyToFetch['routeSegment'] : $relatedSegment;
```

Comment:
typo here - ```entiy``` instead of ```entity```

Reply:
A typo i replicated across the code so it's not breaking. OMG! :) fixed. shame! hehe